fs/remote/blob.go

```diff
@@ -165,20 +185,31 @@ func (b *blob) Cache(offset int64, size int64, opts ...Option) error {
 	fr := b.fetcher
 	b.fetcherMu.Unlock()
 
-	fetchReg := region{floor(offset, b.chunkSize), ceil(offset+size-1, b.chunkSize) - 1}
-	discard := make(map[region]io.Writer)
-	err := b.walkChunks(fetchReg, func(reg region) error {
-		if r, err := b.cache.Get(fr.genID(reg), cacheOpts.cacheOpts...); err == nil {
-			return r.Close() // nop if the cache hits
-		}
-		discard[reg] = ioutil.Discard
-		return nil
-	})
-	if err != nil {
-		return err
+	if b.prefetchChunkSize <= b.chunkSize {
+		return b.fetch(offset, size, fr, &cacheOpts)
+	}
+
+	eg, _ := errgroup.WithContext(context.Background())
+
+	fetchSize := b.chunkSize * (b.prefetchChunkSize / b.chunkSize)
+
+	begin := offset
+	end := offset + size
+
+	for begin < end {
+		start := begin
+
+		eg.Go(func() error {
+			if start+fetchSize > end {
+				return b.fetch(start, end-start, fr, &cacheOpts)
+			}
+			return b.fetch(start, fetchSize, fr, &cacheOpts)
+		})
+
+		begin = begin + fetchSize
 	}
 
-	return b.fetchRange(discard, &cacheOpts)
+	return eg.Wait()
```

Comment:
Could it be simplified like the following?

https://play.golang.org/p/4k-6WwbzlXy

```golang
	end := offset + size
	for i := offset; i < end ; i += fetchSize {
		i, l := i, fetchSize
		if i + l > end {
			l = end - i
		}
		eg.Go(func() error {
			return b.fetch(i, l, fr, &cacheOpts)
		})
	}
	return eg.Wait()
```

Reply:
Ack. I'll update and send out another iteration.